openquake/hazardlib/gsim/kanno_2006.py

```diff
@@ -0,0 +1,517 @@
+    #: observation site and is the hypocentral distance in the case of
+    #: earthquakes for which the fault model is not available." (p. 880)
+    REQUIRES_DISTANCES = set(('rrup',))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, im_type, stddev_types):
+        # pylint: disable=too-many-arguments
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for specification of input and result values.
+
+        Implements the following equations:
+
+        Equation (5) on p. 881 predicts ground motion for shallow events
+        (depth <= 30 km):
+
+        ``log(pre) = a*M + b*X - log(X + d*10^(e*M)) + c + epsilon``
+
+        "where pre is the predicted PGA (cm/sec^2), PGV (cm/sec), or 5%
+        damped response spectral acceleration (cm/sec^2)" (p. 883) and a,
+        b, c and d are tabulated regression coefficients. Note that
+        subscripts on the regression coeffients have been dropped -
+        subscript `1` denoted "shallow" while subscript `2` denoted "deep"
+        - so that the "deep" model of equation (6) can be implemented trivally
+        by changing coefficients and setting d = 0.
+
+        Equation (8) on p. 883 gives the model used for site amplitfication:
+
+        ``G = p*log(VS30) + q``
+
+        Where p and q are tabulated regression coefficients.
+
+        Equation (9) on p. 884 for the ground motion at a given site:
+
+        ``log(pre_G) = log(pre) + G``
+
+        No adjustment of epsilon is made as a function of VS30.
+
+        Note finally that "log represents log_10 in the present study"
+        (p. 880).
+
+        """
+
+        assert im_type.__class__ in self.DEFINED_FOR_INTENSITY_MEASURE_TYPES
+
+        # obtain coefficients for required intensity measure type (IMT)
+        coeffs = self.COEFFS_BASE[im_type].copy()
+        coeffs.update(self.COEFFS_SITE[im_type])
+
+        # obtain IMT-independent coefficients
+        coeffs.update(self.CONSTS)
+
+        # raise warning outside applicable range
+        is_valid = ~((dists.rrup < coeffs['R_valid']) &
+                     (rup.mag > coeffs['M_valid']))
+        if not is_valid.all():
+            warnings.warn('%s used outside applicable range for M=%g at %g km'
+                          % (self.__class__.__name__,
+                             rup.mag, dists.rrup[is_valid][0]))
+
+        # compute bedrock motion, equation (5)
+        log_mean = self._compute_mag_dist_terms(rup, dists, coeffs)
+
+        # make site corrections, equation (9)
+        log_mean += self._compute_site_amplification(sites, coeffs)
+
+        # retrieve standard deviations
+        log_stddevs = self._get_stddevs(coeffs, stddev_types)
+
+        # convert from common to natural logarithm
+        ln_mean = log_mean*np.log(10)
+        ln_stddevs = log_stddevs*np.log(10)
+
+        # convert accelerations from cm/s^2 to g
+        if im_type.__class__.__name__ != 'PGV':
```

Comment:
We have always done `if isinstance(imt, PGV)`. I don't know if one way is better than another, so perhaps Michele can comment